planner/optimize.go

```diff
@@ -211,14 +211,18 @@ func handleInvalidBindRecord(ctx context.Context, sctx sessionctx.Context, stmtN
 
 // isPointGetWithoutDoubleRead returns true when meets following conditions:
 //  1. ctx is auto commit tagged.
-//  2. plan is point get by pk.
+//  2. plan is point get or batch point get by pk.
 func isPointGetWithoutDoubleRead(ctx sessionctx.Context, p plannercore.Plan) bool {
 	if !ctx.GetSessionVars().IsAutocommit() {
 		return false
 	}
 
-	v, ok := p.(*plannercore.PointGetPlan)
-	return ok && v.IndexInfo == nil
+	pointGetPlan, ok := p.(*plannercore.PointGetPlan)
+	if ok {
+		return pointGetPlan.IndexInfo == nil
+	}
+	batchPointGetPlan, ok := p.(*plannercore.BatchPointGetPlan)
```

Comment:
Won't it break the isolation level?

Reply:
It won't break the isolation level if the `BatchPointGet` is constructed by handles.

Reply:
Why? Suppose transaction 1 set (a=1, b=2), transaction 2 set (a=2, b=1), using `MaxTS` may read `a=1, b=1`, is it expected?

Reply:
Yes, you are right, I will fix it.